core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -126,9 +129,15 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
                 log.info("Pulling image " + nativeConfig.builderImage);
                 Process pullProcess = null;
                 try {
-                    pullProcess = new ProcessBuilder(Arrays.asList(containerRuntime, "pull", nativeConfig.builderImage))
-                            .inheritIO()
-                            .start();
+                    final ProcessBuilder pb = new ProcessBuilder(
+                            Arrays.asList(containerRuntime, "pull", nativeConfig.builderImage));
+                    if (processInheritIODisabled.isPresent()) {
+                        // explicitly stream the stdout/stderr of the new process to this process' System.out/System.err
+                        pullProcess = pb.start();
+                        ProcessUtil.streamToSysOutSysErr(pb.start());
+                    } else {
+                        pullProcess = pb.inheritIO().start();
+                    }
```

Comment:
Given that this code is now repeated everywhere the process is launched, how about adding a method to `ProcessInheritIODsiabled` that takes `ProcessBuilder` or arguments necessary to initialize and start it?

Reply:
You are right - this is repeated at more than one place, although there's one place where it deals with it differently. Makes sense to have this in a common method and use it wherever appropriate. I'll update the PR shortly.

Reply:
Hello @aloubyansky, I've moved that logic into `ProcessUtil#launchProcess` and `ProcessUtil#launchProcessStreamStdOut`. PR has now been updated.